scripts/exregional_run_prdgen.sh

```diff
@@ -0,0 +1,366 @@
+#
+
+if [ ${PREDEF_GRID_NAME} = "RRFS_NA_3km" ]; then
+
+DATA=$COMOUT
+DATAprdgen=$DATA/prdgen_${fhr}
+mkdir $DATAprdgen
+
+wgrib2 ${COMOUT}/rrfs.${cycle}.prslev.f${fhr}.grib2 >& $DATAprdgen/prslevf${fhr}.txt
+
+# Create parm files for subsetting on the fly - do it for each forecast hour
+# 10 subpieces for North American grid
+sed -n -e '1,120p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_1.txt
+sed -n -e '121,240p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_2.txt
+sed -n -e '241,360p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_3.txt
+sed -n -e '361,480p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_4.txt
+sed -n -e '481,600p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_5.txt
+sed -n -e '601,720p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_6.txt
+sed -n -e '721,750p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_7.txt
+sed -n -e '751,780p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_8.txt
+sed -n -e '781,880p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_9.txt
+sed -n -e '881,$p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/namerica_10.txt
+
+# 4 subpieces for CONUS and Alaska grids
+sed -n -e '1,250p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/conus_ak_1.txt
+sed -n -e '251,500p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/conus_ak_2.txt
+sed -n -e '501,750p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/conus_ak_3.txt
+sed -n -e '751,$p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/conus_ak_4.txt
+
+# 2 subpieces for Hawaii and Puerto Rico grids
+sed -n -e '1,500p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/hi_pr_1.txt
+sed -n -e '501,$p' $DATAprdgen/prslevf${fhr}.txt >& $DATAprdgen/hi_pr_2.txt
+
+# Create script to execute production generation tasks in parallel using CFP
+echo "#!/bin/bash" > $DATAprdgen/poescript_${fhr}
+echo "export DATA=${DATAprdgen}" >> $DATAprdgen/poescript_${fhr}
+echo "export COMOUT=${COMOUT}" >> $DATAprdgen/poescript_${fhr}
+
+tasks=(4 4 2 2 10)
+domains=(conus ak hi pr namerica)
+count=0
+for i in ${domains[@]}
+do
+  domain=$i
+  for task in $(seq ${tasks[count]})
+  do
+    mkdir -p $DATAprdgen/prdgen_${domain}_${task}
+    echo "$SCRIPTSdir/exregional_run_prdgen_subpiece.sh $fhr $cyc $task $domain ${DATAprdgen} ${COMOUT} &" >> $DATAprdgen/poescript_${fhr}
+  done
+  count=$count+1
+done
+
+echo "wait" >> $DATAprdgen/poescript_${fhr}
+chmod 775 $DATAprdgen/poescript_${fhr}
+
+#-----------------------------------------
+# Execute the script
+#-----------------------------------------
+
+PREP_STEP
+export CMDFILE=$DATAprdgen/poescript_${fhr}
+eval ${RUN_CMD_PRDGEN} ${CMDFILE} ${REDIRECT_OUT_ERR} || print_err_msg_exit "\
+Call to parallel prdgen for forecast hour $fhr returned with non-
+zero exit code."
+POST_STEP
+
+#----------------------------------------
+# reassemble the output grids
+#----------------------------------------
+
+tasks=(4 4 2 2 10)
+domains=(conus ak hi pr namerica)
+count=0
+for i in ${domains[@]}
```

Comment:
It's a little more readable with 

```
for domain in ${domains[@]}
```

Reply:
Yes that looks better, I'll modify both loops to use for domain in ${domains[@]}